### PR TITLE
[PM-147] Import error states usability improvements

### PR DESCRIPTION
**Pull request by djsmith85**
## Type of change

```
- [ ] Bug fix
- [X] New feature development
- [X] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

### Build import-error-dialog
- **apps/web/src/app/tools/import-export/dialog/import-error-dialog.component.html:**  and **apps/web/src/app/tools/import-export/dialog/import-error-dialog.component.ts:** Build import-error-dialog using CL
- **apps/web/src/app/tools/import-export/dialog/index.ts:** Export import-error-dialog.component folder

### Export and use import-error-dialog
- **apps/web/src/app/tools/import-export/import-export.module.ts:** Import and export ImportErrorDialogComponent in module
- **apps/web/src/app/tools/import-export/import.component.ts:** Replace sweetalert2-dialog with DialogService using the new ImportErrorDialogComponent

### BEFORE:
old_formatting_error
old_multiple_errors

### NEW:
new_formatting_error
new_multiple_errors

## Before you submit

- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team
- Ensure that all UI additions follow [WCAG AA requirements](https://contributing.bitwarden.com/contributing/accessibility/)

### CODE REVIEW

**Review by aj-rosado — APPROVED** [review]
LGTM

File: apps/web/src/app/tools/import-export/dialog/import-error-dialog.component.html [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    <bit-table [dataSource]="dataSource">
+      <ng-container header>
+        <tr>
+          <th bitCell>{{ "name" | i18n }}</th>
+          <th bitCell>{{ "description" | i18n }}</th>
+        </tr>
+      </ng-container>
+      <ng-template body let-rows$>
+        <tr bitRow *ngFor="let r of rows$ | async">
+          <td bitCell>{{ r.type }}</td>
```
Comment (Hinton) [inline apps/web/src/app/tools/import-export/dialog/import-error-dialog.component.html:17]: I think type is always empty, should it just be a table with a single column?

File: apps/web/src/app/tools/import-export/dialog/import-error-dialog.component.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
+import { DialogRef, DIALOG_DATA } from "@angular/cdk/dialog";
+import { Component, Inject, OnInit } from "@angular/core";
+
+import { TableDataSource } from "@bitwarden/components";
+
+export interface ErrorList {
+  type: string;
+  message: string;
+}
```
Comment (willmartian) [inline apps/web/src/app/tools/import-export/dialog/import-error-dialog.component.ts:9]: This should be renamed `ErrorListItem` (or something similar) to convey that this is not already an array.

File: apps/web/src/app/tools/import-export/dialog/import-error-dialog.component.html (hunk 1 above)
Comment (djsmith85) [reply apps/web/src/app/tools/import-export/dialog/import-error-dialog.component.html:17]: Yes, the type/name column will for now be empty. This was in agreement to move forward (unblock removal of sweetalert2) and create a follow-up with [PM-3786](https://bitwarden.atlassian.net/browse/PM-3786) to address the missing error types.

**Review by willmartian — DISMISSED** [review]
LGTM 🌈

**Review by Hinton — APPROVED** [review]

**Pull request closed without merging**